File: android/src/main/java/im/zego/zego_express_engine/internal/ZegoPlatformView.java
//
//  android
//  im.zego.zego_express_engine
//
//  Created by Patrick Fu on 2020-04-03.
//  Copyright © 2020 Zego. All rights reserved.
//

package im.zego.zego_express_engine.internal;

import android.content.Context;
import android.view.SurfaceView;
import android.view.View;

import io.flutter.plugin.platform.PlatformView;

public class ZegoPlatformView implements PlatformView {

    private SurfaceView surfaceView;

    ZegoPlatformView(Context context) {
        this.surfaceView = new SurfaceView(context);
        ZegoLog.log("[ZegoPlatformView] [init] surfaceView:%s", this.surfaceView.hashCode());
    }

    public SurfaceView getSurfaceView() {
        return this.surfaceView;
    }

    @Override
    public View getView() {
        return this.surfaceView;
    }

    @Override
    public void dispose() {
        ZegoLog.log("[ZegoPlatformView] [dispose] surfaceView:%s", this.surfaceView.hashCode());
        this.surfaceView = null;
    }
}

File: ios/Classes/internal/ZegoPlatformView.m
//
//  ZegoPlatformView.m
//  Pods-Runner
//
//  Created by lizhanpeng@ZEGO on 2020/3/26.
//  Copyright © 2020 Zego. All rights reserved.
//

#import "ZegoPlatformView.h"
#import "ZegoLog.h"

@interface ZegoPlatformView()

@property (nonatomic, strong, readwrite) ZGView *view;
@property (nonatomic, assign) int64_t viewID;

@end

@implementation ZegoPlatformView

- (instancetype)initWithRect:(CGRect)rect viewID:(int64_t)viewID {
    self = [super init];
    if (self) {
#if TARGET_OS_IPHONE
        _view = [[UIView alloc] initWithFrame:rect];
        _view.backgroundColor = [UIColor blackColor];
#elif TARGET_OS_OSX
        _view = [[NSView alloc] initWithFrame:rect];
        // _view.wantsLayer = YES;
        // _view.layer.backgroundColor = NSColor.clear.cgColor;
#endif
        _viewID = viewID;
    }

    ZGLog(@"[ZegoPlatformView] [init] View:%p", self.view);
    
    return self;
}

- (void)dealloc {
    ZGLog(@"[ZegoPlatformView] [dispose] View:%p", self.view);
}

# pragma mark - Flutter Platform View Delegate

- (ZGView *)view {
    return _view;
}


@end
